### PR TITLE
Cutting a Rod in JAVA

### PR DESCRIPTION
**Pull request by krishna-y2000**
### Have you read the [Contributing Guidelines on Pull Requests](https://github.com/TesseractCoding/NeoAlgo/blob/master/CONTRIBUTING.md#pull-requests)?
Yes
(Write your answer here.)

### Description
I have contributed rod cutting problem. I have done in code in  JAVA.

### Checklist

- [x] I've read the contribution guidelines.
- [x] I've checked the issue list before deciding what to submit.
- [x] I've edited the `README.md` and link to my code.

## Related Issues or Pull Requests

#2489

### CODE REVIEW

File: Java/dp/Rod_Cutting.java [hunk 1]
```diff
@@ -0,0 +1,105 @@
+/*Problem Description
+
+You are given a rod of length N inches and an array of prices that contains prices of all pieces of size smaller than N. 
+Determine the maximum value obtainable by cutting up the rod and selling the pieces.
+
+*/
+
+// importing required package
```
Comment (ankitaggarwal23) [inline Java/dp/Rod_Cutting.java]: Remove the redundant comments

File: Java/dp/Rod_Cutting.java [hunk 2]
```diff
@@ -0,0 +1,105 @@
...
+        for(int i = 0 ; i < m  ; i++)
+            size[i] = i + 1;
+
+        // call solve() method for solving the problem
+        int ans = solve(price , size, m , n);
+        return ans ;
+    }
+    
+    //solve() method will act as solving the problem
+
```
Comment (ankitaggarwal23) [inline Java/dp/Rod_Cutting.java]: Remove the extra one line space

File: Java/dp/Rod_Cutting.java [hunk 3]
```diff
@@ -0,0 +1,105 @@
...
+    int solve(int[] price , int[] size , int m , int n  )
+    {
+
```
Comment (ankitaggarwal23) [inline Java/dp/Rod_Cutting.java]: Remove the extra one line space

File: Java/dp/Rod_Cutting.java [hunk 4]
```diff
@@ -0,0 +1,105 @@
...
+        
+        for(int i = 0 ; i < m + 1; i++)
+        {
+            for(int j = 0 ; j < n + 1 ; j++)
+            {
+                // if size = 0 or there is 0 value for the size
+                if(i == 0 || j ==0 )
+                {
+                    dp[i][j] = 0;
+
```
Comment (ankitaggarwal23) [inline Java/dp/Rod_Cutting.java]: Remove the extra one line space

File: Java/dp/Rod_Cutting.java [hunk 5]
```diff
@@ -0,0 +1,105 @@
...
+                // if size is greater than current max size , no value is taken from price array
+                else
+                {
+                    dp[i][j] = dp[i-1][j];
+                }
+            }
+        }
+        return dp[m][n];
+    }
+
```
Comment (ankitaggarwal23) [inline Java/dp/Rod_Cutting.java]: Remove the extra one line space

File: Java/dp/Rod_Cutting.java [hunk 6]
```diff
@@ -0,0 +1,105 @@
...
+	            int n = sc.nextInt();
+	            int[] arr = new int[n];
+
+                // get values of price 
+	            for (int i = 0; i < n; i++) 
+	            	arr[i] = sc.nextInt();
+
+            //Creating object of another class 
+	            Helper help = new Helper();
+	            System.out.println(help.getValue(arr, n));
```
Comment (ankitaggarwal23) [inline Java/dp/Rod_Cutting.java]: Align your code and comment properly

File: Java/dp/Rod_Cutting.java [hunk 7]
```diff
@@ -0,0 +1,105 @@
...
+	        }
+	    }
+
```
Comment (ankitaggarwal23) [inline Java/dp/Rod_Cutting.java]: Remove the extra one line space

File: Java/dp/Rod_Cutting.java [hunk 8]
```diff
@@ -0,0 +1,105 @@
...
+	 
+}
+
+/* Input
+
+N = 8
+1 5 8 9 10 17 17 20
+
+Output
+22
```
Comment (ankitaggarwal23) [inline Java/dp/Rod_Cutting.java]: Kindly use display message to read and display the I/P & O/P

File: Java/dp/Rod_Cutting.java [hunk 9]
```diff
@@ -0,0 +1,105 @@
...
+
+Explanation
+The maximum value we get is 22 by cutting in two pieces of lengths 2 and 6 . So , 22 is answer.
+
+Time Complexity : O(N^2)
+Space Complexity : O(N)
+*/
+
+
+
```
Comment (ankitaggarwal23) [inline Java/dp/Rod_Cutting.java]: Remove the extra two line space

**Review by ankitaggarwal23 — CHANGES_REQUESTED** [review]
Also cross check your code output it is showing infinite loop.

**Comment by krishna-y2000** [comment]
@ankitaggarwal23  hii , 
i have applied the requested changes . 
[My running code](https://ideone.com/5bAt6F)
Please merge it.

File: Java/dp/Rod_Cutting.java [hunk 10]
```diff
@@ -0,0 +1,88 @@
+/*Problem Description
+You are given a rod of length N inches and an array of prices that contains prices of all pieces of size smaller than N. 
+Determine the maximum value obtainable by cutting up the rod and selling the pieces.
+*/
```
Comment (ankitaggarwal23) [inline Java/dp/Rod_Cutting.java:4]: Add a new line after */

File: Java/dp/Rod_Cutting.java [hunk 11]
```diff
@@ -0,0 +1,88 @@
...
+import java.util.Scanner;
```
Comment (ankitaggarwal23) [inline Java/dp/Rod_Cutting.java:6]: Add a new line after header file

**Review by ankitaggarwal23 — CHANGES_REQUESTED** [review]
Do the previously marked changes as well.

**Comment by krishna-y2000** [comment]
@ankitaggarwal23  changes are applied . Have a look.

**Comment by krishna-y2000** [comment]
@ankitaggarwal23 hii,
if you find any spare time then please review my pr. This might be the last check.
@HarshCasper

**Review by ankitaggarwal23 — CHANGES_REQUESTED** [review]
Do the changes mentioned

**Comment by krishna-y2000** [comment]
> Do the changes mentioned

@ankitaggarwal23  I have updated  the code.Have a look .

File: Java/dp/Rod_Cutting.java (hunk 2 above)
Comment (krishna-y2000) [reply Java/dp/Rod_Cutting.java]: I have applied the change in commit, see the final file.

File: Java/dp/Rod_Cutting.java (hunk 3 above)
Comment (krishna-y2000) [reply Java/dp/Rod_Cutting.java]: Applied the change in commit .

Comment (krishna-y2000) [reply Java/dp/Rod_Cutting.java]: See the final file. Applied the change.

File: Java/dp/Rod_Cutting.java (hunk 4 above)
Comment (krishna-y2000) [reply Java/dp/Rod_Cutting.java]: Ok,Applied the change.

File: Java/dp/Rod_Cutting.java (hunk 6 above)
Comment (krishna-y2000) [reply Java/dp/Rod_Cutting.java]: Ok, applied in last commit.

File: Java/dp/Rod_Cutting.java (hunk 7 above)
Comment (krishna-y2000) [reply Java/dp/Rod_Cutting.java]: See the final file. Applied the change .

File: Java/dp/Rod_Cutting.java (hunk 8 above)
Comment (krishna-y2000) [reply Java/dp/Rod_Cutting.java]: Applied changes in commit.

File: Java/dp/Rod_Cutting.java (hunk 9 above)
Comment (krishna-y2000) [reply Java/dp/Rod_Cutting.java]: Ok, applied in last commit.

**Review by HarshCasper — APPROVED** [review]

**Pull request merged**